x-pack/plugins/observability_solution/slo/public/hooks/use_fetch_slo_health.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ALL_VALUE, FetchSLOHealthResponse, SLOWithSummaryResponse } from '@kbn/slo-schema';
+import { useQuery } from '@tanstack/react-query';
+import { useKibana } from '../utils/kibana_react';
+import { sloKeys } from './query_key_factory';
+
+export interface UseFetchSloHealth {
+  data: FetchSLOHealthResponse | undefined;
+  isLoading: boolean;
+  isError: boolean;
+}
+
+export interface Params {
+  list: SLOWithSummaryResponse[];
+}
+
+export function useFetchSloHealth({ list }: Params): UseFetchSloHealth {
+  const { http } = useKibana().services;
+  const payload = list.map((slo) => ({
+    sloId: slo.id,
+    sloInstanceId: slo.instanceId ?? ALL_VALUE,
+  }));
+
+  const { isLoading, isError, data } = useQuery({
+    queryKey: sloKeys.health(payload),
+    queryFn: async ({ signal }) => {
+      try {
+        const response = await http.post<FetchSLOHealthResponse>(
+          '/internal/observability/slos/_health',
```

Comment:
We have to limit to the provided list because fetching all SLOs at once would be too much _in some case_. Same way we don't return all SLOs in the find API.

But what we can do instead, is use directly the transform stats endpoint from this diagnostic tool with the `slo-*` id, this will return in one request all the SLO transform stats. (I think there is still a limit on this API, like 1000)
Then the diagnostic tool could filter & transform the result to keep only the health.status part of it, or return the payload as is.

This API is already available and GET.